### PR TITLE
[ShaderGraph][2022.1] Select best match based on relevance across all items returned by searcher

### PR DESCRIPTION
**Pull request by jessebarker**
### Purpose of this PR
This PR fixes https://fogbugz.unity3d.com/f/cases/1366058/

The last searcher integration update could leave us not actually visiting all of the returned search results, and only used the length of the results to select a best match if there was no exact match.  This could yield some surprising results (e.g., searching for "blend" would actually yield "lerp" as the selection).  These changes ensure that we visit each of the search results once, and scores the results based on exact match by name and synonym, as well as partial match by name and synonym (match by name is preferred over match by synonym).

---
### Testing status
Manual testing according to the repro case in the bug as well as using several other duplicated synonyms (e.g. "derive").

[PR ShaderGraph on trunk](https://unity-ci.cds.internal.unity3d.com/job/8914134)

---
### Comments to reviewers
Notes for the reviewers you have assigned.

### CODE REVIEW

**Review by bencloward — APPROVED** [review]
Results look good.  It's selecting the node with the search term in the name now over items with the search term in the synonyms.  Thanks for doing that!

**Comment by Nightmask3** [comment]
https://user-images.githubusercontent.com/13033542/134249446-5b7d3529-b375-496b-8fd2-703f9e5a797c.mp4

Given that we seem to have a lot of search results in the Searcher that are split by whitespaces, and those whitespaces can interfere with the search process as shown above, could we format the search results to remove the whitespaces before we do comparisons?

I'd think that removing the whitespaces from the searchQuery would also help towards this 🤔

**Review by Nightmask3 — COMMENTED** [review]
Saw that you already addressed the unused variables issue, would like to see if we can fix the edge case I took a video of

File: com.unity.shadergraph/Editor/Drawing/SearchWindowAdapter.cs [hunk 1]
```diff
@@ -18,50 +18,117 @@ public SearchWindowAdapter(string title) : base(title)
...
+        {
+            // Scoring Criteria:
+            // - Exact name match is most preferred.
+            // - Partial name match is next.
+            // - Exact synonym match is next.
+            // - Partial synonym match is next.
+            // - No match is last.
+            int score = 0;
+
+            bool isSynonymMatch = false;
```
[the remark this answers is not in the document]
Comment (jessebarker) [reply com.unity.shadergraph/Editor/Drawing/SearchWindowAdapter.cs]: Yup.  The CI complained about them (they are warnings in the console, but in the CI they turn into errors.  Removed in latest branch HEAD.

**Review by Nightmask3 — APPROVED** [review]
Discussed with Jesse about the edge case I left a comment about, came to the conclusion that we don't want to diverge the search filtration from the actual search terms that the user provided and the results the internal Searcher logic is matching against, that's a whole other can of worms potentially

**Comment by jessebarker** [comment]
Thanks to @cdxntchou for the tweaks to the multi-term scoring!

**Pull request merged**